### ISSUE TITLE
Error 500 after updating NC 16 to NC 17 r2

### ISSUE THREAD

**Issue by petrk94** [nextcloud/server#17338]
Hello,
Im facing a bug from my NC 16 updated to NC 17 r2 instance.
After a successful updating process, it wanted to load the web interface and caused the Error 500.
No access is possible.
In the php error log, I got this information:

[Mon Sep 30 11:40:59 2019] [warn] [client 94.223.33.225] mod_fcgid: stderr: PHP Fatal error:  Declaration of OCA\\GroupFolders\\Versions\\VersionsBackend::getVersionFile(OCP\\IUser $user, OCP\\Files\\FileInfo $sourceFile, int $revision): OCP\\Files\\File must be compatible with OCA\\Files_Versions\\Versions\\IVersionBackend::getVersionFile(OCP\\IUser $user, OCP\\Files\\FileInfo $sourceFile, $revision): OCP\\Files\\File in /var/www/virtual/petrk/nextcloud.xxx/apps/groupfolders/lib/Versions/VersionsBackend.php on line 37
[Mon Sep 30 11:41:01 2019] [warn] [client 66.102.8.52] mod_fcgid: stderr: PHP Fatal error:  Declaration of OCA\\GroupFolders\\Versions\\VersionsBackend::getVersionFile(OCP\\IUser $user, OCP\\Files\\FileInfo $sourceFile, int $revision): OCP\\Files\\File must be compatible with OCA\\Files_Versions\\Versions\\IVersionBackend::getVersionFile(OCP\\IUser $user, OCP\\Files\\FileInfo $sourceFile, $revision): OCP\\Files\\File in /var/www/virtual/petrk/nextcloud.xxx/apps/groupfolders/lib/Versions/VersionsBackend.php on line 37

I hope you can help me to restore it to bring it back to work.

Greets

**Comment by rth9** [comment]
Probably a better way, but:
`sudo mv /var/www/virtual/petrk/nextcloud.xxx/apps/groupfolders /var/www/virtual/petrk/nextcloud/apps/groupfolders.old`
Then you can proceed and it gets reinstalled from what I see. I'm still looking. Probably can delete the newly created ".old" folder after the upgrade.

**Comment by kesselb** [comment]
> Probably can delete the newly created ".old" folder after the upgrade.

Yes. 

cc @nextcloud/server-triage the issue itself is already fixed by groupfolders 5.x.

**Comment by skjnldsv** [comment]
@kesselb so I'm guessing we just did not released groupfolders on time?

**Comment by kesselb** [comment]
Hmm. I don't think so. 5.0.0 (with the relevant changes) is out for 25 days.

**Comment by Schmuuu** [comment]
Hi @kesselb @skjnldsv 

Users are running into this issue when updating to NC17 stable. 
https://help.nextcloud.com/t/web-updater-from-nc-16-0-4-to-17-doesnt-work-with-group-folders-use-command-line/61110
This issue arises when group folders app was enabled before the update.

The big issue is, that group folders app in version 5.0 is not compatible to NC16 and therefore users cannot update to group folder 5.x before updating to NC17. In return all users who didn't disable the app before the update, will get stuck in HTTP 500.

For users without ssh access this may become a serious issue.

**Comment by petrk94** [comment]
@Schmuuu 
how I can solve it?

**Comment by Schmuuu** [comment]
I posted a warning in the forum:
https://help.nextcloud.com/t/nc17-update-disable-group-folders-app-before-performing-the-update-of-your-server/61112/2

To solve the issue:
- login to your server via ssh
- run the occ command “upgrade”:
   `sudo -u <web user> php /path/to/nc/occ upgrade`

**Comment by petrk94** [comment]
Here the log (system work now, issue solved by this solution):

> php occ upgrade
The current PHP memory limit is below the recommended value of 512MB.
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
Set log level to debug
Turned on maintenance mode
Updating database schema
Updated database
Disabled incompatible app: files_texteditor
Disabled incompatible app: groupfolders
Updating <accessibility> ...
Updated <accessibility> to 1.3.0
Updating <federatedfilesharing> ...
Updated <federatedfilesharing> to 1.7.0
Updating <files_pdfviewer> ...
Updated <files_pdfviewer> to 1.6.0
Updating <files_rightclick> ...
Updated <files_rightclick> to 0.14.2
Updating <files_videoplayer> ...
Updated <files_videoplayer> to 1.6.0
Updating <gallery> ...
Updated <gallery> to 18.4.0
Updating <logreader> ...
Updated <logreader> to 2.2.0
Updating <password_policy> ...
Updated <password_policy> to 1.7.0
Updating <privacy> ...
Updated <privacy> to 1.1.0
Updating <provisioning_api> ...
Updated <provisioning_api> to 1.7.0
Updating <recommendations> ...
Updated <recommendations> to 0.5.0
Updating <serverinfo> ...
Updated <serverinfo> to 1.7.0
Updating <support> ...
Updated <support> to 1.0.1
Updating <twofactor_backupcodes> ...
Updated <twofactor_backupcodes> to 1.6.0
Updating <updatenotification> ...
Updated <updatenotification> to 1.7.0
Updating <viewer> ...
Updated <viewer> to 1.1.0
Updating <lookup_server_connector> ...
Updated <lookup_server_connector> to 1.5.0
Updating <oauth2> ...
Updated <oauth2> to 1.5.0
Updating <files> ...
Updated <files> to 1.12.0
Updating <activity> ...
Updated <activity> to 2.10.1
Updating <cloud_federation_api> ...
Updated <cloud_federation_api> to 1.0.0
Updating <dav> ...
Fix broken values of calendar objects

 Done
    0/0 [>---------------------------]   0%
Updated <dav> to 1.13.0
Updating <files_external> ...
Updated <files_external> to 1.8.0
Updating <files_sharing> ...
Updated <files_sharing> to 1.9.0
Updating <files_trashbin> ...
Updated <files_trashbin> to 1.7.0
Updating <files_versions> ...
Updated <files_versions> to 1.10.0
Updating <sharebymail> ...
Updated <sharebymail> to 1.7.0
Updating <workflowengine> ...
Updated <workflowengine> to 1.7.0
Updating <admin_audit> ...
Updated <admin_audit> to 1.7.0
Updating <comments> ...
Updated <comments> to 1.7.0
Updating <firstrunwizard> ...
Updated <firstrunwizard> to 2.6.0
Updating <notifications> ...
Updated <notifications> to 2.5.0
Updating <systemtags> ...
Updated <systemtags> to 1.7.0
Updating <theming> ...
Updated <theming> to 1.8.0
Checking for update of app accessibility in appstore
Checked for update of app "accessibility" in appstore
Checking for update of app activity in appstore
Checked for update of app "activity" in appstore
Checking for update of app admin_audit in appstore
Checked for update of app "admin_audit" in appstore
Checking for update of app cloud_federation_api in appstore
Checked for update of app "cloud_federation_api" in appstore
Checking for update of app comments in appstore
Checked for update of app "comments" in appstore
Checking for update of app dav in appstore
Checked for update of app "dav" in appstore
Checking for update of app federatedfilesharing in appstore
Checked for update of app "federatedfilesharing" in appstore
Checking for update of app files in appstore
Checked for update of app "files" in appstore
Checking for update of app files_external in appstore
Checked for update of app "files_external" in appstore
Checking for update of app files_fulltextsearch_tesseract in appstore
Update app files_fulltextsearch_tesseract from appstore
Checked for update of app "files_fulltextsearch_tesseract" in appstore
Checking for update of app files_pdfviewer in appstore
Checked for update of app "files_pdfviewer" in appstore
Checking for update of app files_rightclick in appstore
Update app files_rightclick from appstore
Checked for update of app "files_rightclick" in appstore
Checking for update of app files_sharing in appstore
Checked for update of app "files_sharing" in appstore
Checking for update of app files_trashbin in appstore
Checked for update of app "files_trashbin" in appstore
Checking for update of app files_versions in appstore
Checked for update of app "files_versions" in appstore
Checking for update of app files_videoplayer in appstore
Checked for update of app "files_videoplayer" in appstore
Checking for update of app firstrunwizard in appstore
Checked for update of app "firstrunwizard" in appstore
Checking for update of app fulltextsearch in appstore
Update app fulltextsearch from appstore
Checked for update of app "fulltextsearch" in appstore
Checking for update of app gallery in appstore
Checked for update of app "gallery" in appstore
Checking for update of app logreader in appstore
Checked for update of app "logreader" in appstore
Checking for update of app lookup_server_connector in appstore
Checked for update of app "lookup_server_connector" in appstore
Checking for update of app notifications in appstore
Checked for update of app "notifications" in appstore
Checking for update of app oauth2 in appstore
Checked for update of app "oauth2" in appstore
Checking for update of app password_policy in appstore
Checked for update of app "password_policy" in appstore
Checking for update of app privacy in appstore
Checked for update of app "privacy" in appstore
Checking for update of app provisioning_api in appstore
Checked for update of app "provisioning_api" in appstore
Checking for update of app recommendations in appstore
Checked for update of app "recommendations" in appstore
Checking for update of app serverinfo in appstore
Checked for update of app "serverinfo" in appstore
Checking for update of app sharebymail in appstore
Checked for update of app "sharebymail" in appstore
Checking for update of app support in appstore
Checked for update of app "support" in appstore
Checking for update of app systemtags in appstore
Checked for update of app "systemtags" in appstore
Checking for update of app theming in appstore
Checked for update of app "theming" in appstore
Checking for update of app twofactor_backupcodes in appstore
Checked for update of app "twofactor_backupcodes" in appstore
Checking for update of app updatenotification in appstore
Checked for update of app "updatenotification" in appstore
Checking for update of app viewer in appstore
Checked for update of app "viewer" in appstore
Checking for update of app workflowengine in appstore
Checked for update of app "workflowengine" in appstore
Checking for update of app files_texteditor in appstore
Checked for update of app "files_texteditor" in appstore
Checking for update of app groupfolders in appstore
Update app groupfolders from appstore
Checked for update of app "groupfolders" in appstore
Starting code integrity check...
Finished code integrity check
Update successful
Turned off maintenance mode
Reset log level

**Comment by gwolf** [comment]
I suppose there are plans underway for Group Folders to be supported in NC17 by the time of the official release, right? Our use case certainly depends on GF working :-\

**Comment by Schmuuu** [comment]
@gwolf The groupfolders app is supported in NC17 already. The only issue here is, that the groupfolders app in version 5.x is not compatible with NC16 and in version 4.x is not compatible with NC17. So in a way an upgrade path is missing.

The workaround therefore is:
- disable group folders app
- update NC from 16.x to 17.0
- check that group folders app has been upgraded to version 5.x (if not, update manually)
- enable group folders app again

**Comment by gwolf** [comment]
OK, got it. Thank you very much!

**Comment by DaCryptor** [comment]
> To solve the issue:
> 
>     * login to your server via ssh
> 
>     * run the occ command “upgrade”:
>       `sudo -u <web user> php /path/to/nc/occ upgrade`

Great "workaround" for a lot of people running NextCloud in a shared-hosting environment...

To be honest, I see the NextCloud-team releasing more and more features very fast, but also more and more errors occur (even in the basic functions), and I am losing trust in the quality of this software.

**Comment by kesselb** [comment]
```
  'app_install_overwrite' => 
  array (
    0 => 'groupfolders',
  ),
```

Do you have something like above in your `config.php`?

**Comment by Schmuuu** [comment]
No. I only saw this getting added:
```
'app_install_overwrite' => 
array (
0 => 'tasks',
1 => 'files_readmemd',
2 => 'polls',
3 => 'apporder',
),
```

**Comment by kyrofa** [comment]
> The only issue here is, that the groupfolders app in version 5.x is not compatible with NC16 and in version 4.x is not compatible with NC17. So in a way an upgrade path is missing.

Wait, that's the exact reason we in the snap held off on the upgrade to v16, the calendar app ended up doing the same thing. However, that situation was fixed with #16507, where Nextcloud now disables incompatible apps before the upgrade, and updates them/re-enables them afterward. Did that break?

**Comment by kesselb** [comment]
> where Nextcloud now disables incompatible apps before the upgrade, and updates them/re-enables them afterward. Did that break?

For me it works :confused:  16.0.5 with groupfolders 4.x to 17.0.0 will disable groupfolders during upgrade. I can reproduce the fatal error `OCP\Files\File must be compatible with OCA\Files_Versions\Versions\IVersionBackend::getVersionFile` with "enable untested app". We have to dig deeper :see_no_evil:

**Comment by My1** [comment]
curious question regarding this: is the possible to knock out an app without a commandline, like in the database or some config file and could that help shared hosting users?

**Comment by petrk94** [comment]
> > To solve the issue:
> > ```
> > * login to your server via ssh
> > 
> > * run the occ command “upgrade”:
> >   `sudo -u <web user> php /path/to/nc/occ upgrade`
> > ```
> 
> Great "workaround" for a lot of people running NextCloud in a shared-hosting environment...
> 
> To be honest, I see the NextCloud-team releasing more and more features very fast, but also more and more errors occur (even in the basic functions), and I am losing trust in the quality of this software.

@DaCryptor Im running my NC instance on a shared host, the whole thing worked too without need of sudo. Just php /path/to/nc/occ upgrade and it worked

**Comment by jthacker48** [comment]
> I posted a warning in the forum:
> https://help.nextcloud.com/t/nc17-update-disable-group-folders-app-before-performing-the-update-of-your-server/61112/2
> 
> To solve the issue:
> 
> * login to your server via ssh
> * run the occ command “upgrade”:
>   `sudo -u <web user> php /path/to/nc/occ upgrade`

I must be making a mistake when trying to run the command as I'm getting "can't open web: no such file".  

I'm running NC as Docker in Unraid.  I'm putting in the following (exactly):

sudo -u <web user> php /config/www/nextcloud/occ upgrade

I'm sure it's something dumb so I apologize in advance.

**Comment by tigernero79** [comment]
https://github.com/nextcloud/server/issues/17378

**Comment by tigernero79** [comment]
in my opinion, with a web-based update, the check of the group folder version is not done or updated. reason why we have error 500. on systems where we have access ssh we solve with command via terminal "occ upgrade", the verification is done because it is successfully updated.

**Comment by rschr** [comment]
The malfunction reported in this issue just happened to me in trying to upgrading from 16.0.6 to 17.0.1 through the browser-based upgrader. Reloading the web page in the browser to complete the upgrade left me with a http 500 error and the above mentioned PHP error connected with group folders in the logs. (Performing `occ upgrade` on the server's command line finished the job, though.)

**Comment by moritzps** [comment]
> Probably a better way, but:
> `sudo mv /var/www/virtual/petrk/nextcloud.xxx/apps/groupfolders /var/www/virtual/petrk/nextcloud/apps/groupfolders.old`
> Then you can proceed and it gets reinstalled from what I see. I'm still looking. Probably can delete the newly created ".old" folder after the upgrade.

This is the only way I found to fix it on a shared hosting without SSH access (I only have SFTP access). I'm still uncertain how to proceed now, because I can not update groupfolders from the NC17 web-based app updater and am afraid I'd loose the files I have in my group folder when removing the app and reinstalling it via the app store.
![groupfolders_NC17-v4](https://user-images.githubusercontent.com/1819792/68999227-1063b600-08be-11ea-8242-af3a22bae842.png)

**Comment by ph00lt0** [comment]
Just updated on the stable channel and had to disable the group folders manually. To be able to update the installation. It was complaining about integrity of the update.secret.

**Comment by DudeBroChill** [comment]
My NC is also on shared hosting with no ssh access. I was able to finish the upgrade by using ftp into my server and going in to the apps folder and renaming the folder “groupfolders” to “STOPgroupfolders”. I then reloaded the NC page and the installation continued... then it warned me about all the incompatible apps.

**Comment by kyrofa** [comment]
> We have to dig deeper 

@kesselb any progress here?

**Comment by kyrofa** [comment]
I'm noticing something interesting that might be related. If I install Nextcloud 16.0.5 with version 0.15.1 of the Right click app, and then try to update to v17 (based on the daily tarball), the update log looks like this:

```
Set log level to debug
Turned on maintenance mode
Updating database schema
Updated database
Disabled incompatible app: files_texteditor
Updating <accessibility> ...
Updated <accessibility> to 1.3.0
Updating <federatedfilesharing> ...
Updated <federatedfilesharing> to 1.7.0
Updating <files_pdfviewer> ...
Updated <files_pdfviewer> to 1.6.0
Updating <files_rightclick> ...
Updated <files_rightclick> to 0.14.2
Updating <files_videoplayer> ...
Updated <files_videoplayer> to 1.6.0
Updating <gallery> ...
Updated <gallery> to 18.4.0
Updating <logreader> ...
Updated <logreader> to 2.2.0
Updating <password_policy> ...
Updated <password_policy> to 1.7.0
Updating <privacy> ...
Updated <privacy> to 1.1.0
Updating <provisioning_api> ...
Updated <provisioning_api> to 1.7.0
Updating <recommendations> ...
Updated <recommendations> to 0.5.0
Updating <serverinfo> ...
Updated <serverinfo> to 1.7.0
Updating <support> ...
Updated <support> to 1.0.1
Updating <survey_client> ...
Updated <survey_client> to 1.5.0
Updating <twofactor_backupcodes> ...
Updated <twofactor_backupcodes> to 1.6.0
Updating <federation> ...
Updated <federation> to 1.7.0
Updating <lookup_server_connector> ...
Updated <lookup_server_connector> to 1.5.0
Updating <oauth2> ...
Updated <oauth2> to 1.5.0
Updating <files> ...
Updated <files> to 1.12.0
Updating <activity> ...
Updated <activity> to 2.10.1
Updating <cloud_federation_api> ...
Updated <cloud_federation_api> to 1.0.0
Updating <dav> ...
Fix broken values of calendar objects
 Starting ...
    0/0 [>---------------------------]   0%
Updated <dav> to 1.13.0
Updating <files_sharing> ...
Updated <files_sharing> to 1.9.0
Updating <files_trashbin> ...
Updated <files_trashbin> to 1.7.0
Updating <files_versions> ...
Updated <files_versions> to 1.10.0
Updating <sharebymail> ...
Updated <sharebymail> to 1.7.0
Updating <workflowengine> ...
Updated <workflowengine> to 1.7.0
Updating <comments> ...
Updated <comments> to 1.7.0
Updating <firstrunwizard> ...
Updated <firstrunwizard> to 2.6.0
Updating <nextcloud_announcements> ...
Updated <nextcloud_announcements> to 1.6.0
Updating <notifications> ...
Updated <notifications> to 2.5.0
Updating <systemtags> ...
Updated <systemtags> to 1.7.0
Checking for update of app accessibility in appstore
Checked for update of app "accessibility" in appstore
Checking for update of app activity in appstore
Checked for update of app "activity" in appstore
Checking for update of app bruteforcesettings in appstore
Checked for update of app "bruteforcesettings" in appstore
Checking for update of app cloud_federation_api in appstore
Checked for update of app "cloud_federation_api" in appstore
Checking for update of app comments in appstore
Checked for update of app "comments" in appstore
Checking for update of app dav in appstore
Checked for update of app "dav" in appstore
Checking for update of app federatedfilesharing in appstore
Checked for update of app "federatedfilesharing" in appstore
Checking for update of app federation in appstore
Checked for update of app "federation" in appstore
Checking for update of app files in appstore
Checked for update of app "files" in appstore
Checking for update of app files_pdfviewer in appstore
Checked for update of app "files_pdfviewer" in appstore
Checking for update of app files_rightclick in appstore
Update app files_rightclick from appstore
Checked for update of app "files_rightclick" in appstore
Checking for update of app files_sharing in appstore
Checked for update of app "files_sharing" in appstore
Checking for update of app files_trashbin in appstore
Checked for update of app "files_trashbin" in appstore
Checking for update of app files_versions in appstore
Checked for update of app "files_versions" in appstore
Checking for update of app files_videoplayer in appstore
Checked for update of app "files_videoplayer" in appstore
Checking for update of app firstrunwizard in appstore
Checked for update of app "firstrunwizard" in appstore
Checking for update of app gallery in appstore
Checked for update of app "gallery" in appstore
Checking for update of app logreader in appstore
Checked for update of app "logreader" in appstore
Checking for update of app lookup_server_connector in appstore
Checked for update of app "lookup_server_connector" in appstore
Checking for update of app nextcloud_announcements in appstore
Checked for update of app "nextcloud_announcements" in appstore
Checking for update of app notifications in appstore
Checked for update of app "notifications" in appstore
Checking for update of app oauth2 in appstore
Checked for update of app "oauth2" in appstore
Checking for update of app password_policy in appstore
Checked for update of app "password_policy" in appstore
Checking for update of app privacy in appstore
Checked for update of app "privacy" in appstore
Checking for update of app provisioning_api in appstore
Checked for update of app "provisioning_api" in appstore
Checking for update of app recommendations in appstore
Checked for update of app "recommendations" in appstore
Checking for update of app serverinfo in appstore
Checked for update of app "serverinfo" in appstore
Checking for update of app sharebymail in appstore
Checked for update of app "sharebymail" in appstore
Checking for update of app support in appstore
Checked for update of app "support" in appstore
Checking for update of app survey_client in appstore
Checked for update of app "survey_client" in appstore
Checking for update of app systemtags in appstore
Checked for update of app "systemtags" in appstore
Checking for update of app twofactor_backupcodes in appstore
Checked for update of app "twofactor_backupcodes" in appstore
Checking for update of app viewer in appstore
Checked for update of app "viewer" in appstore
Checking for update of app workflowengine in appstore
Checked for update of app "workflowengine" in appstore
Checking for update of app files_texteditor in appstore
Checked for update of app "files_texteditor" in appstore
Starting code integrity check...
Finished code integrity check
Update successful
Turned off maintenance mode
Reset log level
```

Then I visited Nextcloud in my browser and saw this:

![Screenshot from 2019-11-19 16-27-10](https://user-images.githubusercontent.com/946674/69198413-76805100-0ae9-11ea-99a7-23a8415ea5fc.png)

So... the upgrade didn't work? Hmm. I tried running it again:

```
Set log level to debug
Turned on maintenance mode
Updating database schema
Updated database
Updating <files_rightclick> ...
Updated <files_rightclick> to 0.15.1
Checking for update of app accessibility in appstore
Checked for update of app "accessibility" in appstore 
Checking for update of app activity in appstore
Checked for update of app "activity" in appstore 
Checking for update of app bruteforcesettings in appstore
Checked for update of app "bruteforcesettings" in appstore 
Checking for update of app cloud_federation_api in appstore
Checked for update of app "cloud_federation_api" in appstore 
Checking for update of app comments in appstore
Checked for update of app "comments" in appstore 
Checking for update of app dav in appstore
Checked for update of app "dav" in appstore 
Checking for update of app federatedfilesharing in appstore
Checked for update of app "federatedfilesharing" in appstore 
Checking for update of app federation in appstore
Checked for update of app "federation" in appstore 
Checking for update of app files in appstore
Checked for update of app "files" in appstore 
Checking for update of app files_pdfviewer in appstore
Checked for update of app "files_pdfviewer" in appstore 
Checking for update of app files_rightclick in appstore
Checked for update of app "files_rightclick" in appstore 
Checking for update of app files_sharing in appstore
Checked for update of app "files_sharing" in appstore 
Checking for update of app files_trashbin in appstore
Checked for update of app "files_trashbin" in appstore 
Checking for update of app files_versions in appstore
Checked for update of app "files_versions" in appstore 
Checking for update of app files_videoplayer in appstore
Checked for update of app "files_videoplayer" in appstore 
Checking for update of app firstrunwizard in appstore
Checked for update of app "firstrunwizard" in appstore 
Checking for update of app gallery in appstore
Checked for update of app "gallery" in appstore 
Checking for update of app logreader in appstore
Checked for update of app "logreader" in appstore 
Checking for update of app lookup_server_connector in appstore
Checked for update of app "lookup_server_connector" in appstore 
Checking for update of app nextcloud_announcements in appstore
Checked for update of app "nextcloud_announcements" in appstore 
Checking for update of app notifications in appstore
Checked for update of app "notifications" in appstore 
Checking for update of app oauth2 in appstore
Checked for update of app "oauth2" in appstore 
Checking for update of app password_policy in appstore
Checked for update of app "password_policy" in appstore 
Checking for update of app privacy in appstore
Checked for update of app "privacy" in appstore 
Checking for update of app provisioning_api in appstore
Checked for update of app "provisioning_api" in appstore 
Checking for update of app recommendations in appstore
Checked for update of app "recommendations" in appstore 
Checking for update of app serverinfo in appstore
Checked for update of app "serverinfo" in appstore 
Checking for update of app sharebymail in appstore
Checked for update of app "sharebymail" in appstore 
Checking for update of app support in appstore
Checked for update of app "support" in appstore 
Checking for update of app survey_client in appstore
Checked for update of app "survey_client" in appstore 
Checking for update of app systemtags in appstore
Checked for update of app "systemtags" in appstore 
Checking for update of app text in appstore
Checked for update of app "text" in appstore 
Checking for update of app twofactor_backupcodes in appstore
Checked for update of app "twofactor_backupcodes" in appstore 
Checking for update of app viewer in appstore
Checked for update of app "viewer" in appstore 
Checking for update of app workflowengine in appstore
Checked for update of app "workflowengine" in appstore 
Starting code integrity check...
Finished code integrity check
Update successful
Turned off maintenance mode
Reset log level
```

And NOW it works! What... ? So I try doing it a third time:

```
Nextcloud is already latest version
```

Why didn't that happen the second time? Why didn't the first upgrade take despite seemingly encountering no errors?

**Comment by kyrofa** [comment]
The above actually happens from 16.0.5 to 16.0.6 as well; perhaps some breakage was backported.

**Comment by kyrofa** [comment]
Given that v0.13.0 of files_rightclick supports v17, I'm not sure why it's forcing the update to happen before continuing. Note that the first upgrade updates it to 0.14.2, and only the second gets it to 0.15.1.

**Comment by ph00lt0** [comment]
I tried to do some investigation on it. If you run the web updater from 16 to 17 you get the empty page. After this I revisit the updater page, this page complains about the integrity as well and suggests entering the plaintext value of the updater secret. (hashed version stored in the web.config file). 
When updating the update secret and entering the plain text value as well it response with: Token invalid. Would this mean that the groupfolders somehow use a different key?

**Comment by kesselb** [comment]
cc @nextcloud/server-triage 

https://github.com/nextcloud/server/pull/16145 regarding the `files_rightclick` app. Probably the updater is confused somehow because it's shipped. I don't know if updating shipped apps via appstore is possible. 

As workaround for the snap update I would suggest to remove the groupfolders app and install it after the update. `occ app:remove --keep-data groupfolders` will remove the app but keep the (user)data. If the app was installed $? = 0 and if not 1.

**Comment by moritzps** [comment]
> As workaround for the snap update I would suggest to remove the groupfolders app and install it after the update. `occ app:remove --keep-data groupfolders` will remove the app but keep the (user)data. If the app was installed $? = 0 and if not 1.

Do you know if there is a possibility to keep the data when using the web UI? I do not have ssh access and would like to update the groupfolders app but can’t because it requires NC16 to be updated (which does not make any sense). See screenshot above. 
I‘m afraid if I delete the app now and reinstall afterwards all data is gone.

**Comment by skjnldsv** [comment]
> I don't know if updating shipped apps via appstore is possible.

It is not possible because of file integrity check

**Comment by royord** [comment]
> > Probably a better way, but:
> > `sudo mv /var/www/virtual/petrk/nextcloud.xxx/apps/groupfolders /var/www/virtual/petrk/nextcloud/apps/groupfolders.old`
> > Then you can proceed and it gets reinstalled from what I see. I'm still looking. Probably can delete the newly created ".old" folder after the upgrade.
> 
> This is the only way I found to fix it on a shared hosting without SSH access (I only have SFTP access). I'm still uncertain how to proceed now, because I can not update groupfolders from the NC17 web-based app updater and am afraid I'd loose the files I have in my group folder when removing the app and reinstalling it via the app store.
> ![groupfolders_NC17-v4](https://user-images.githubusercontent.com/1819792/68999227-1063b600-08be-11ea-8242-af3a22bae842.png)

Also resolved my issue.

**Comment by DudeBroChill** [comment]
I have not received any negative repercussions from just renaming the groupfolders folder using ftp and then finishing the re-stall. Worked perfectly, and then groupfolders just installed its self afterwards. I’ve had zero data loss. This seems to be the easy around around this.

John Stutsman

**Comment by AstroGD** [comment]
Hi,
I have the same issue but now Im stuck because Im not able to do php occ upgrade.

I only get this error:

```
An unhandled exception has been thrown:
Exception: Environment not properly prepared. in /cloud.*******.**/lib/private/Console/Application.php:166
Stack trace:
#0 /cloud.*******.**/console.php(96): OC\Console\Application->loadCommands(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#1 /cloud.*******.**/occ(11): require_once('/cloud.*******....')
```

Does anyone know how to resolve this without data loss?

**Comment by skjnldsv** [comment]
@AstroGD please give us the logged json error in your nextcloud log file. We're missing info on your comment

**Comment by AstroGD** [comment]
I don't know where the log file is located. But I figured out how to solve the update issue myself:
I reinstalled the backup created by the updater and then followed the steps on this issue by deactivating group-folders, updating and the re-enabling the plugin after installing the newest version of it. So far there are no issues and I installed another update without issues today. Maybe the logfiles are still there - If so I would happily send them to you after I know where they are located.
